apps/nccl/src/nccl.cu

```diff
@@ -644,3 +647,38 @@ NCCL_API ncclResult_t ncclGroupEnd() {
   // Do nothing
   return ncclSuccess;
 }
+
+NCCL_API ncclResult_t ncclCommRegister(const ncclComm_t, void*, size_t, void**) {
+  // TODO: Implementation
+  return ncclSuccess;
+}
+
+NCCL_API ncclResult_t ncclCommDeregister(const ncclComm_t, void*) {
+  // TODO: Implementation
+  return ncclSuccess;
+}
+
+ncclResult_t ncclMemAlloc(void** ptr, size_t size) {
+  // Allocate memory using mscclpp::allocSharedPhysicalCuda
+  auto rawPtr = mscclpp::allocSharedPhysicalCuda<char>(size);
```

Comment:
The name is confusing; the returned value is a shared pointer, now a raw pointer

Reply:
> The name is confusing; the returned value is a shared pointer, now a raw pointer

Change the name rawPtr to sharedPtr in latest commit.